seacatauth/external_login/authentication/providers/oauth2.py

```
@@ -319,10 +320,16 @@ async def _get_verified_claims(self, id_token, expected_nonce: str | None = None
 				claims = json.loads(id_token.claims)
 				return claims
 			except jwcrypto.jws.InvalidJWSSignature:
-				L.error("Invalid ID token signature.", struct_data={"provider": self.Type})
+				AuditLogger.error("Authentication failed", struct_data={
+					"provider": self.Type,
+					"reason": "Invalid ID token signature",
+				})
 				raise ExternalLoginError("Invalid ID token signature.")
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)